### DISCUSSION TITLE
How to use "children" of a custom component in solid-elements?

### DISCUSSION THREAD

**Discussion by onx2** [solidjs/solid#2140] [Q&A] [answered]
I can't figure out how to pass along the children from the consumer into my custom component. Does anyone know how to do this?

consumer:
```
<wb-button variant="asdf">button</wb-button>
```

my custom element:
```
customElement(
  "wb-button",
  { variant: "one" },
  (props: WbButtonProps, { element }) => {
    return <button>{element.children}</button>;
  }
);
```

what I end up with:
<img width="706" alt="image" src="https://github.com/solidjs/solid/assets/23039725/cce5ad2a-ecb8-4290-afa5-20d784ab4c19">

**Comment by onx2** [comment, marked as answer]
I think I figured this out though I don't find it very ergonomic so if anyone has a better approach, I would love to hear it.

```ts
customElement(
  "wb-button",
  { variant: "one" },
  (props: WbButtonProps, { element }) => {
    return (
      <button>
        <style>{style}</style>
        <For each={element.childNodes}>{(child) => child}</For>
      </button>
    );
  }
);
```

```ts
<wb-button variant="asdf">
        <span>button</span>
        asdf
</wb-button>
```
<img width="333" alt="image" src="https://github.com/solidjs/solid/assets/23039725/e8f8f3a4-e169-417b-892f-75129286e164">

**Comment by akirchhoff-modular** [comment]
I think the expected way to do this is to use a `slot` element, e.g.:

```jsx
customElement(
  "wb-button",
  { variant: "one" },
  (props: WbButtonProps) => {
    return <button><slot /></button>;
  }
);
```

This works for me, but moreover, it works when Solid context providers are in use, while Solid context providers do not work with the `<For each={element.childNodes}>` approach. I found this solution from the example on https://github.com/solidjs/solid/issues/785.
